### DISCUSSION TITLE
`.` in syntax for shell globbing

### DISCUSSION THREAD

**Discussion by ahillio** [xonsh/xonsh#4703] [General]
I installed xonsh and started testdriving it and I was disappointed to realize the shell globbing require a `.` here...

```py
for filename in `.*`: # a period that wouldn't be required in normal shell
  do stuff

# and the following...
for filename in `*`: # this doesn't work (without `.`)
  do stuff
```
Why is this?
I imagine it's for a particular reason... and can't be changed?
It would be so much nicer to be consistent with more common shells and not require this extra character.
It also makes me wonder what other inconsistencies exist?

**Comment by gforsyth** [comment]
Backticks perform regex-style globs, for your standard glob, you can prepend with a `g`

```
(base) gil@kaltenecker ~/tmp 🐚  `.*`
['dir with a space',
 'index.html',
 'install',
 'xonsh-0.10.1',
 'xonsh-0.10.1.tar.gz',
 'xonsh-0.11.0',
 'xonsh-0.11.0.tar.gz',
 'xonsh-x86_64.AppImage']
(base) gil@kaltenecker ~/tmp 🐚  g`*`
['dir with a space',
 'index.html',
 'install',
 'xonsh-0.10.1',
 'xonsh-0.10.1.tar.gz',
 'xonsh-0.11.0',
 'xonsh-0.11.0.tar.gz',
 'xonsh-x86_64.AppImage']
```

**Reply by ahillio** [reply]
Nice, thanks!  Is there a way to do globbing without the backticks?

**Reply by gforsyth** [reply]
you can `glob.glob('*')` as you would in Python, but that might be less convenient.

**Reply by ahillio** [reply]
I might be hoping for "a magical blending of the two" when there's only a "reality-constrained blending"...?

I'm surprised to see that `*` on its own (what's this do?) initializes a git repo - huh?

**Reply by ahillio** [reply]
Of course reality's a reasonable thing to be constrained by.
Thanks for chiming in here!!!
